misc/peer-store/src/connection_store.rs

```diff
@@ -0,0 +1,91 @@
+use libp2p_core::{ConnectedPoint, PeerId};
+use libp2p_swarm::ConnectionId;
+use std::collections::{HashMap, HashSet};
+
+/// Events emitted by the connection tracker behaviour.
+#[derive(Debug, Clone)]
+pub enum Event {
+    /// A peer connected (first connection established).
+    PeerConnected {
+        peer_id: PeerId,
+        connection_id: ConnectionId,
+        endpoint: ConnectedPoint,
+    },
+
+    /// A peer disconnected (last connection closed).
+    PeerDisconnected {
+        peer_id: PeerId,
+        connection_id: ConnectionId,
+    },
```

Comment:
I don't think we need these events. We already have the swarm-events for established and closed connections, that include the info how many other connections to that peer exist.